src/main/java/ui/components/pickers/MilestonePickerState.java

```diff
@@ -30,82 +34,244 @@ public MilestonePickerState(List<PickerMilestone> milestones, String userInput)
+     * @param userInput
+     * @param currentMilestones
+     * @param limit
+     * @param allMilestones
+     */
+    private List<PickerMilestone> getLimitedMatchesFromPreviousInput(String userInput,
+                                                                     List<PickerMilestone> currentMilestones,
+                                                                     int limit, List<PickerMilestone> allMilestones) {
+        List<PickerMilestone> matchesForPreviousInput = new ArrayList<>();
+        String curUserInput = userInput;
+        while (matchesForPreviousInput.size() < limit && curUserInput.length() > 0) {
+            List<PickerMilestone> totalList = new ArrayList<>(currentMilestones);
+            totalList.addAll(matchesForPreviousInput);
+
+            List<PickerMilestone> matchingMilestones = getNewMatchingMilestones(curUserInput, totalList,
+                                                                                allMilestones);
+            matchesForPreviousInput.addAll(matchingMilestones);
+
+            curUserInput = curUserInput.substring(0, curUserInput.length() - 1);
+        }
+
+        return matchesForPreviousInput.stream()
+                .limit(limit)
+                .collect(Collectors.toList());
     }
 
-    private Optional<PickerMilestone> getFirstMatchingMilestone(List<PickerMilestone> milestoneList) {
-        return milestoneList.stream()
-                .filter(milestone -> !milestone.isMatching())
-                .findFirst();
+    /**
+     * Gets a limited number of milestones from {@code milestones} whose isMatching() returns true
+     *
+     * Assumption: {@code limit} is not negative
+     * @param limit non-negative integer
+     * @param milestones milestones to get the matching milestones from
+     */
+    private List<PickerMilestone> getLimitedMatchingMilestones(int limit, List<PickerMilestone> milestones) {
+        return milestones.stream()
+                .filter(PickerMilestone::isMatching)
+                .limit(limit)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Gets milestones from {@code allMilestones} which match {@code userInput} and are not already
+     * included in {@code currentMilestones}
+     *
+     * @param userInput
+     * @param currentMilestones
+     * @param allMilestones
+     */
+    private List<PickerMilestone> getNewMatchingMilestones(String userInput, List<PickerMilestone> currentMilestones,
+                                                           List<PickerMilestone> allMilestones) {
+        String[] curUserInputWords = userInput.split(" ");
+
+        List<PickerMilestone> matchingMilestones = allMilestones;
+        for (String word : curUserInputWords) {
+            matchingMilestones = matchingMilestones.stream()
+                    .filter(milestone -> isMatchingQuery(milestone, word))
+                    .collect(Collectors.toList());
+        }
```

Comment:
I don't think that is a bug and is intended. Multiple query words will only match milestones that match all query words (this allows the user to progressively narrow down the list of milestones). In the above example, it will match milestones that has `Ver` and `1.3` e.g. `Ver 1.3` and `Ver 1.3.99`.